x-pack/plugins/security_solution/server/search_strategy/security_solution/factory/risk_score/kpi/query.kpi_risk_score.dsl.ts

```diff
@@ -24,12 +25,16 @@ export const buildKpiRiskScoreQuery = ({
       aggs: {
         risk: {
           terms: {
-            field: 'risk.keyword',
+            field:
+              entity === RiskScoreEntity.user ? RiskScoreFields.userRisk : RiskScoreFields.hostRisk,
           },
           aggs: {
             unique_entries: {
               cardinality: {
```

Comment:
@machadoum , Since we are already aggregating by a field such as `RiskScoreFields.userRisk`  or `RiskScoreFields.hostRisk`, wouldn't cardinality be always 1 `(i.e. unique_entries)` for that particular field? Is it intensional? See screenshot below.

![image](https://user-images.githubusercontent.com/7485038/189337418-e3e7501e-3021-40d8-adc5-fcc1bb6f4a15.png)

Do you think below was the intention? Apologies if I have missed something.

```json
 {
  "aggs": {
    "risk": {
      "terms": {
        "field": "user.risk.calculated_score_norm"
      }
    },
    "unique_entries": {
      "cardinality": {
        "field": "user.risk.calculated_score_norm"
      }
    }
  }
}
```

Reply:
There is a mistake in the code above, `unique_entries.cardinality.field` should be `user.name` or `host.name`. If the fields of the term aggregation and cardinality are the same, the result would always be `1`, but that is not the case. Here is a sample of the query that runs:
```
    "aggs": {
      "risk": {
        "terms": {
          "field": "host.risk.calculated_level"
        },
        "aggs": {
          "unique_entries": {
            "cardinality": {
              "field": "host.name"
            }
          }
        }
    }
```

The `terms` aggregation groups all entries by risk severity, and `cardinality` counts how many users and hosts exist for each severity.

Reply:
Okay... this makes sense. My bad 😰